### ISSUE TITLE
HR Report for Parent and Child accounts

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#1287]
In GitLab by @kdjstudios on Dec 13, 2018, 15:52

**Submitted by:** Kyle

**Helpdesk:** NA

**Server:** Internal

**Client/Site:** VCC (All?)

**Account:** parent and child accounts

**Issue:** 

It would appear that the HR report is not working for parent and child accounts as I would expect. I discovered this while researching #1289 

- Parent account: The parent account is not displaying the HR report for any of the ussage for its child accounts. In #1289 all child accounts are be billing up into the parent account, so I would expect to be able to see the the HR of all child accounts in the parent account.

- Child Accounts: We do not have access to the HR report on child account? I had already opened a separate ticket #1155 to address this concern.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Dec 31, 2018, 02:29

@kdjstudios 

we did some R&D and found same report can work for child accounts,  given the resource id is present and different than it's parent account.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 31, 2018, 08:55

@smasih @tim.traylor 

Thank you for the update. I think we should enable the HR report for child accounts and simply put a 'warning' that displays on top of the report. Something along the lines of "This is a child account and is only tracking usage information from the XXXX Resource ID. Parent account billing and usage may vary accordingly."

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 2, 2019, 06:15

@kdjstudios 

We dug into the code and found the same report is work for the child accounts when enabled hr report link for the child account.

For now, we have changed the code of the account page to enabled hr report for the child account.

We have tested on the dev server. Please test it on dev server.
 
Here the parent and child account that we have to test.  
Parent account - "DEMO-54"   https://dev.sabilling.com/accounts/5b658fd2cf85c153b800000a  
child account - "DM-AC-1"  https://dev.sabilling.com/accounts/59424316cf85c1844000051d

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 2, 2019, 06:15

added 3h of time spent at 2019-01-02

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 2, 2019, 09:22

@amishra @smasih @pchaudhary @tim.traylor @abakshi @rchauhan 

Why was this worked on? There was no request approved for estimate, no estimate provided, and no approval for work. 

This was marked for New/Incoming so this could be discussed with Tim and Gary. Marking back for New/Incoming so it can be handled properly.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 7, 2019, 12:47

Reviewed with @tim.traylor, we will want to discuss why this was worked on without an estimate or approval during tomorrows meeting.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Jan 16, 2019, 09:00

@rchauhan Hi Richie, did we get any info on how this was worked on without an estimate or us putting it in the for development bucket?  Thx

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 16, 2019, 09:27

@amishra 

Discussed with @tim.traylor today and since this has already been developed we will want to test it and release it accordingly.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jan 17, 2019, 03:04

@rchauhan @tim.traylor 

We have to do the R&D before the estimate or even post a comment because we need to be sure that the information we are passing is correct.

So, there was some R&D done and found out that we don't need to do any development on this if we can just unhide the link on UI. 

and Ashish tested the theory out and that was all required on this. 

Bottom line issue was resolved during the R&D. (total time spent on this from beginning 2.50h)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 24, 2019, 10:26

@smasih @amishra 

Please post this into a dev server so we may test it. Please update the labels.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jan 24, 2019, 23:01

assigned to @amishra

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 25, 2019, 04:32

@kdjstudios 

Thank you for the update.

We have set up the branch for your testing. Please test it and let us know our valuable feedback.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 25, 2019, 09:27

@amishra @smasih @pchaudhary @tim.traylor @rchauhan @abakshi 

Which server is this on? Please be sure to update the labels. This is only causing delays and costing extra time for everyone.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 27, 2019, 22:11

@kdjstudios 

Thanks

We'll keep in mind from the next time.

Please test it on the dev server.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jan 27, 2019, 22:13

@amishra 

check twice that all the labels are updated before forwarding it for testing.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 28, 2019, 08:45

@amishra @smasih @tim.traylor 

After trying to login to the dev server multiple times. I realized that every time I navigate to https://dev.sabilling.com/users/sign_in it forwards me to Answernet production of https://answernet.sabilling.com/users/sign_in

- What is going on? 
 - *I reached out to @tim.traylor to see if this is happening to him too. EDIT: Tim was also able to confirm this was happening to him. He believes this is a redirect issue.*

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 28, 2019, 10:51

@kdjstudios 

Thank you for the update.

Please check it again.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 28, 2019, 11:19

@smasih @amishra 

I was able to confirm that the HR link is now visable for child accounts. However, I was unable to have it display any metrics for billing cycles that the child accounts had usage on. I also tried to process a new billing cycle, in case there was actually no historical data. However I receive this error. 

![image](/uploads/befecbe4803a171d913761e47ec518b6/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 28, 2019, 22:25

@kdjstudios 

Thank you for the update.

We have resolved the issue. Please test it.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 29, 2019, 10:13

@amishra @smasih @pchaudhary @tim.traylor @rchauhan @abakshi 

Please explain how this was tested and confirmed working?

I just tested this on DEV in the Demo site. I uploaded a usage file and processed a billing cycle. All accounts regardless if they are a parent account or a child account are no longer showing anything in the HR report.

![image](/uploads/6ce5885f121f641ddd5a3d869a43e5f3/image.png)

![image](/uploads/c8fb8c76b52eb3992ccf1867b09a4315/image.png)

Also: 
- Now when I edit a child account, the "parent" drop down is not populating. So if I save the account, it then un-links it from the parent account.
- Parent accounts no longer have access to any links other then the HR report.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 30, 2019, 05:40

@kdjstudios 

Thank you for the update.



> Please explain how this was tested and confirmed working?

We have tested with production data on the dev server. Please test it.

* Select account number **T-062-INB6833** under **Toronto, Canada site** Site. 
* select the billing cycle **Usage Start: November 25, 2018, Usage end: December 24, 2018, Invoice date: January 01, 2019, Cycle: Master**

* If the account is child account of the **NCSM - Aviem** account.

![image](/uploads/a745727b1a5f6109794a015fb7218621/image.png) 
![image](/uploads/99ed01d88721a2cc91b4a7e6af6e4168/image.png)


 



* If the same account is a parent account.

![image](/uploads/0f65ea254091b2010a7dc3f34b62f108/image.png)
![image](/uploads/7f89a011c52baf9b5b9dd94c181ec678/image.png)



Hope this help.

thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 30, 2019, 09:49

@amishra @pchaudhary @smasih 

Please look at site Demo and any child account for the Demo account 1 (Child Resource IDs: DM-12060,061-2036,DM-54,DM-31255,DM-7293,DM-81853). These accounts all have usage on the usage file. [MERGE_DEMO_BILLING.csv](/uploads/db248023bfbb9fc54291de0a9b457cb1/MERGE_DEMO_BILLING.csv) 

I created a new billing cycle and processed the above usage and finalized the billing cycle. For each child account the HR report is not working, we are getting the result below. Even the Parent account is no longer working. 

![image](/uploads/057164a9376a7a96e974e014cfb9e348/image.png)

Also @amishra Please only use one 'PL' label on a ticket. using multiple 'pl' labels only causes confusion on the status of the ticket.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jan 30, 2019, 22:26

@amishra 

follow this it might help you: 

```ruby
cdr = Cdr.where(billing_cycle_id: "selected billing cycle id").first
records = cdr.where(cdr_id: cdr._id)
hr_account = records.where(resource_id: "resource id of account in question")
```
* The values/data we get in hr_account should match the data we have on report.
* If there is nothing in the hr_account then:
 * either account's resource id is changed or
 * there was no usage uploaded for that account in that billing cycle.
* Lastly the switch label on site/billing cycle should be same as when usage was uploaded for that account

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 30, 2019, 22:42

@kdjstudios @smasih 

thank you for the update.


The switch was changed that is why hr report not showing.

We changed the switch with "VCC".

Please test it.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jan 31, 2019, 24:14

assigned to @kdjstudios and unassigned @amishra

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 31, 2019, 12:05

@amishra @smasih 

Thank you, this is now working. Marking this for announcement.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 4, 2019, 10:57

@tim.traylor 

Will we want to release this at the same time we release #1155 ?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 12, 2019, 10:03

removed assignee

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Feb 16, 2019, 04:48

@kdjstudios 

We have successfully released on both of the production servers.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 25, 2019, 13:30

closed
